cli/magic-config.ts

```diff
@@ -170,22 +170,42 @@ async function processCreateOptions(options: any): Promise<void> {
         "Do you want to deploy a private website? I.e only accessible in VPC",
       initial: options.privateWebsite || false,
     },
+    {
+      type: "confirm",
+      name: "customPublicDomain",
+      message:
+        "Do you want to provide a custom domain name and corresponding certificate arn for the public website ?",
+      initial: options.customPublicDomain || false,
+      skip(): boolean {
+        return (this as any).state.answers.privateWebsite ;
+      },
+    },
     {
       type: "input",
       name: "certificate",
-      message: "ACM certificate ARN",
+      message(): string {
+        if ((this as any).state.answers.customPublicDomain) {
+          return "ACM certificate ARN with custom domain for public website. Note that the certificate must resides in us-east-1";
+        }
+        return "ACM certificate ARN";
+      },
       initial: options.certificate,
       skip(): boolean {
-        return !(this as any).state.answers.privateWebsite;
+        return !(this as any).state.answers.privateWebsite && !(this as any).state.answers.customPublicDomain;
       },
     },
     {
       type: "input",
       name: "domain",
-      message: "Domain for private website",
+      message(): string {
```

Comment:
What about asking for Route53 Zone Id too so we can create the [A record with CDK directly](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_route53-readme.html)?

Reply:
That's a good point.
Most of the time Route53 public zone is centralized in an AWS Account. 
This does not limits this possibility (local or central).

I propose to create another PR for this, 
[Domain Names and Certificates](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudfront-readme.html#domain-names-and-certificates)
with a migration to [CloudFrontWebDistribution API](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudfront-readme.html#cloudfrontwebdistribution-api)

Reply:
Agreed